src/fidocadj/FidoFrame.java

```diff
@@ -824,32 +824,41 @@ public void showPrefs()
         repaint();
     }
 
-    /** Set the current zoom to fit
-    */
+    /**
+     Adjusts the zoom level of the drawing to fit the entire content within ..
+     the visible area.
+     Before applying the zoom, it checks if there are any primitives with ..
+     negative coordinates and translates them if necessary to ensure the ..
+     entire drawing is visible.
+     */
     public void zoomToFit()
```

Comment:
I don't think it is the right place to call the function that normalize the coordinates. zoomToFit() should not modify the drawing.
I think it should not be possible to *move* elements so that they do not have negative coordinates. This should only be applied to selected elements as those are the ones that the user expects to be changed.

Reply:
But what if the user had moved a primitive outside the visible area and doesn't remember doing it? The other alternative is that we shouldn't allow elements to be moved outside the drawing area.  
Anyway, we had already discussed this: https://github.com/DarwinNE/FidoCadJ/issues/219, and from what I understood, you had decided on this solution.

Reply:
I read again #219 and it is my fault, indeed at the time I said that it was reasonable to put it in the zoomToFit().
Thinking again, I still consider it is preferable to avoid that zoomToFit() modifies the drawing. I think the user should see a change when they _already_ are doing an action to modify the drawing.